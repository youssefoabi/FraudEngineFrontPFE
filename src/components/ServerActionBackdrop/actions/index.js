export const PENDING_REQUEST = 'PENDING_REQUEST';
export const RESET_PENDING_REQUEST = 'RESET_PENDING_REQUEST';
