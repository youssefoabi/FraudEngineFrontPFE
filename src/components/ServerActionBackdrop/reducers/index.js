import pendingActionNavigation from './navigationReducer';

export default { pendingActionNavigation };
