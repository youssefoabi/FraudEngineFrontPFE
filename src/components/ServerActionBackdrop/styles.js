export default {
  backdrop: {
    zIndex: 10,
    color: '#FFF',
  },
};
