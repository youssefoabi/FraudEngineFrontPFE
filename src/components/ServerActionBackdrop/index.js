import { Backdrop, CircularProgress, makeStyles } from '@material-ui/core';
import { pathOr } from 'ramda';
import { useSelector } from 'react-redux';
import React from 'react';

import styles from './styles';

const useStyles = makeStyles(() => styles);

const ServerActionBackdrop = () => {
  const pendingRequest = useSelector(pathOr(false, ['navigation', 'pendingRequest', 'pendingRequest']));

  const classes = useStyles();

  return (
    <Backdrop className={classes.backdrop} open={pendingRequest}>
      <CircularProgress />
    </Backdrop>
  );
};

export default ServerActionBackdrop;
