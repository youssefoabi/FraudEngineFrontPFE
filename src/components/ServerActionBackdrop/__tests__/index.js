import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import ServerActionBackdrop from '../';

jest.mock('react-redux', () => ({
  useSelector: () => ({}),
}));

describe(ServerActionBackdrop.name, () => {
  itRendersCorrectly(<ServerActionBackdrop />);
});
