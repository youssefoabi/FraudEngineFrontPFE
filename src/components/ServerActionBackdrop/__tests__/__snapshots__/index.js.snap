// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerActionBackdrop renders correctly 1`] = `
<WithStyles(ForwardRef(Backdrop))
  className="makeStyles-backdrop-1"
  open={Object {}}
>
  <WithStyles(ForwardRef(CircularProgress)) />
</WithStyles(ForwardRef(Backdrop))>
`;
